coverage/hare_coverage.py

```diff
@@ -0,0 +1,319 @@
+                directory_path: str = sub_dir + sep + directory
+                if self.omit_dir(directory_path):
+                    continue
+                result_list.extend([ele for ele in self.search_dir(directory_path) if ele not in result_list])
+                if self.callback(directory):
+                    if directory_path not in result_list:
+                        result_list.append(directory_path)
+        return result_list
+
+
+class Utility:
+
+    """Here is the utility class which has static utility helper methods."""
+
+    def run_cmd(self, command: str, delimiter: Optional[str] = None, max_split: int = -1) -> None:
+        """
+        Execute command and return the result.
+        :param command: Command in string format
+        :param delimiter: delimiter (default is space)
+        :param max_split: maximum number split
+        :return: tuple(stdout, stderr)
+        """
+        logging.debug(f"Executing Command : {command!a}")
+        process: Popen[str] = Popen(command.split(sep=delimiter, maxsplit=max_split),
+                                    stdout=PIPE,
+                                    stderr=PIPE)
+        out, err = process.communicate()
+        if process.returncode:
+            raise Exception(
+                f'Command {command!a} exited with error code {process.returncode}.\n'
+                f'Command output: {err.decode("utf-8")}')
+
+        logging.debug(f'Output:\n{out.decode("utf-8")}')
+
+    def is_exist(self, dir_path: str) -> bool:
+        """
+        Check for the existance of the file or directory.
+        :param dir_path: file or directory path
+        :return: bool
+        """
+        return path.exists(dir_path)
+
+    def is_inside_hare(self) -> None:
+        """
+        Check current working directory is hare home directory or not.
+        if not then raise the exception.
+        :return: None
+        """
+        if getcwd().endswith(f"{sep}hare") is False:
+            logging.error("Not running a script from hare home directory.")
+            raise InvalidDirectoryPath("Not running a script from hare home directory.")
+
+
+class Strategy(ABC):
+
+    """Abstract base class for coverage tool."""
+
+    @abstractmethod
+    def cleanup(self) -> None:
+        """
+        Cleaning up old coverage reports.
+        :return: None
+        """
+        pass
+
+    @abstractmethod
+    def run_ut(self) -> None:
+        """
+        Execute unit testcases.
+        :return: None
+        """
+        pass
+
+    @abstractmethod
+    def run_coverage(self) -> None:
+        """
+        Running coverage tool over UTs.
+        :return: None
+        """
+        pass
+
+    @abstractmethod
+    def check_report(self) -> None:
+        """
+        Validate the report generated
+        :return: None
+        """
+        pass
+
+
+class PythonCoverage(Strategy):
+
+    # This is source to test directory map. It will be used run coverage.

```

Comment:
s/It will be used run coverage./It will be used to run coverage.

Reply:
Done